test/SILOptimizer/redundant_load_elim.sil

```diff
@@ -1190,3 +1190,30 @@ bb2(%a : $*Int):
   dealloc_stack %4 : $*Int
   return %r : $Int
 } 
+
+// Make sure that the store is forwarded to the load, ie. the load is
+// eliminated. That's correct as the stored value can't be changed by the
+// callee as it's passed with @in_guaranteed.
+sil @test_rle_in_guaranteed_sink : $@convention(thin) (Int) -> ()
+sil @test_rle_in_guaranteed_callee : $@convention(thin) (@in_guaranteed Int) -> ()
+
+// CHECK-LABEL: sil @test_rle_in_guaranteed_entry
+sil @test_rle_in_guaranteed_entry : $@convention(thin) (@in Int) -> () {
+bb0(%0 : $*Int):
+  %value_raw = integer_literal $Builtin.Int64, 42
+  %value = struct $Int (%value_raw : $Builtin.Int64)
+  store %value to %0 : $*Int
+
+  %f_callee = function_ref @test_rle_in_guaranteed_callee : $@convention(thin) (@in_guaranteed Int) -> ()
+  %r1 = apply %f_callee(%0) : $@convention(thin) (@in_guaranteed Int) -> ()
+
+  // CHECK-NOT: load
+  %value_again = load %0 : $*Int
+
+  %f_sink = function_ref @test_rle_in_guaranteed_sink : $@convention(thin) (Int) -> ()
+  // CHECK: [[VAL:%.*]] = struct $Int
```

Comment:
@eeckstein 

 - `op.Index = 1`
 - `AI->...->getNumParams` = 1
 - `V` =   `%1 = argument of bb0 : $*AnyIterator<τ_0_0>    // user: %3`
 - `*AI` =  `%8 = apply %7<AnyIterator<τ_0_0>>(%0, %2) : $@convention(method) <τ_0_0 where τ_0_0 : Sequence, τ_0_0 == τ_0_0.Iterator> (@in_guaranteed τ_0_0) -> @out τ_0_0`